src/component/contents/exploration/DraftEditorBlockNode.react.js

```diff
@@ -0,0 +1,370 @@
+   * be the case if the user has pressed `RETURN` or pasted some content, since
+   * programatically creating these new blocks and setting the DOM selection
+   * will miss out on the browser natively scrolling to that position.
+   *
+   * To replicate native behavior, if the block overlaps the selection state
+   * on mount, force the scroll position. Check the scroll state of the scroll
+   * parent, and adjust it to align the entire block to the bottom of the
+   * scroll parent.
+   */
+  componentDidMount(): void {
+    const selection = this.props.selection;
+    const endKey = selection.getEndKey();
+    if (!selection.getHasFocus() || endKey !== this.props.block.getKey()) {
+      return;
+    }
+
+    const blockNode = ReactDOM.findDOMNode(this);
+    const scrollParent = Style.getScrollParent(blockNode);
+    const scrollPosition = getScrollPosition(scrollParent);
+    let scrollDelta;
+
+    if (scrollParent === window) {
+      const nodePosition = getElementPosition(blockNode);
+      const nodeBottom = nodePosition.y + nodePosition.height;
+      const viewportHeight = getViewportDimensions().height;
+      scrollDelta = nodeBottom - viewportHeight;
+      if (scrollDelta > 0) {
+        window.scrollTo(
+          scrollPosition.x,
+          scrollPosition.y + scrollDelta + SCROLL_BUFFER,
+        );
+      }
+    } else {
+      invariant(
+        blockNode instanceof HTMLElement,
+        'blockNode is not an HTMLElement',
+      );
+      const blockBottom = blockNode.offsetHeight + blockNode.offsetTop;
+      const scrollBottom = scrollParent.offsetHeight + scrollPosition.y;
+      scrollDelta = blockBottom - scrollBottom;
+      if (scrollDelta > 0) {
+        Scroll.setTop(
+          scrollParent,
+          Scroll.getTop(scrollParent) + scrollDelta + SCROLL_BUFFER,
+        );
+      }
+    }
+  }
+
+  render(): React.Node {
+    const {
+      block,
+      blockRenderMap,
+      blockRendererFn,
+      blockStyleFn,
+      contentState,
+      decorator,
+      editorKey,
+      editorState,
+      customStyleFn,
+      customStyleMap,
+      direction,
+      forceSelection,
+      selection,
+      tree,
+    } = this.props;
+
+    let children = null;
+
+    if (block.children.size) {
+      children = block.children.reduce((acc, key) => {
+        const offsetKey = DraftOffsetKey.encode(key, 0, 0);
+        const child = contentState.getBlockForKey(key);
+        const customConfig = getCustomRenderConfig(child, blockRendererFn);
+        const Component = customConfig.CustomComponent || DraftEditorBlockNode;
+        const {Element, wrapperTemplate} = getDraftRenderConfig(
+          child,
+          blockRenderMap,
+        );
+        const elementProps = getElementPropsConfig(
+          child,
+          editorKey,
+          offsetKey,
+          blockStyleFn,
+          customConfig,
+        );
+        const childProps = {
+          ...this.props,
+          tree: editorState.getBlockTree(key),
```

Comment:
I hate to pass the editorState through in order to use it just once. Not sure how to extract what we need here though.

Reply:
I could pass just the `editorState.getBlockTree` method reference, what do you reckon ?

Reply:
Sounds good - passing a bound function like that is a bit unconventional, but I can't think of any strong reason not to.